ArticlesApp/ArticlesApp/ArticlesViewController.swift

```diff
@@ -0,0 +1,76 @@
+import UIKit
+
+class ArticlesViewController: UIViewController {
+
+    @IBOutlet weak var loadingView: UIView!
+    @IBOutlet var articlesTableView: UITableView!
+
+    var articles = [Article]()
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        loadSampleArticles()
+        setupView()
+    }
+
+    private func setupView() {
+        articlesTableView.dataSource = self
+        articlesTableView.delegate = self
+        articlesTableView.isHidden = true
+        loadingView.isHidden = false
+    }
+
+    private func loadSampleArticles() {
+        let manager = DataManager()
+        manager.fetchArticles { result in
```

Comment:
Every time you call a method that has a completion block like this and calls `self` from it, the app will copy the whole context of your current object to the block. So at this point, `self` has a reference to the completion block and the completion block has a reference to `self` (reference cycle). You need to break this cycle to avoid memory leaking. You can do that by declaring how `self` should be handle inside this block. You have 2 main options: `weak` or `unowned`. The first means that `self` will be interpreted as an optional inside the block (because if `self` dies, this pointer will be set to `nil`. The latter means you have 100% guarantee that `self` will always exist. Which means it won't be an optional anymore but an implicit unwrapped optional, meaning that if you call `self` and it does not exists, it will crash in runtime.

So for me the rule of thumb is: is this method sync or async? If it's async, `weak`. If it's sync (tap on a button for example), then `unowned`.

The declaration comes in the format:

```swift
manager.fetchArticles { [weak self] result in
   // your code here but now using `self?` when referencing the view controller
}
```